src/server/components/auth/authenticated-user.ts

```diff
@@ -50,6 +43,38 @@ export default class AuthenticatedUser {
     return lists ?? [];
   }
 
+  static calculateSortOrder(customSortOrder?: Record<string, string>): Array<Record<string, string>> {
+    const defaultSortOrder: Array<Record<string, string>> = [
+      {
+        country: "asc",
+      },
+      {
+        type: "asc",
+      },
+    ];
+
+    if (!customSortOrder) return defaultSortOrder;
+
+    const queryParamToColumnNames = {
+      administrators: "admins",
+      serviceProviders: "live",
+      annualReview: "isOverdue",
+    };
+
+    type QueryParamKeys = keyof typeof queryParamToColumnNames;
+
+    const formatCustomSortOrder = Object.entries(customSortOrder).map(([key, value]) => {
```

Comment:
I think this will be simplified if we just use object instead of array. I think it's only valuable if you need to `AND` or use `OR`.

the type `ListsForDashboardOrderByWithRelationInput` looks like 

```
  export type ListsForDashboardOrderByWithRelationInput = {
    listId?: SortOrder
    type?: SortOrder
    country?: SortOrder
    admins?: SortOrder
    live?: SortOrder
    actionNeeded?: SortOrder
    isOverdue?: SortOrder
    nextAnnualReviewStartDate?: SortOrder
    lastAnnualReviewStartDate?: SortOrder
    jsonData?: SortOrder
  }
```